### DISCUSSION TITLE
Auto-add issues from multiple repos or organization wide to a project table<title>

### DISCUSSION THREAD

**Discussion by kathy-phet** [community/community#47803] [Projects and Issues]
### Select Topic Area

Product Feedback

### Body

Right now you can only set up an "auto-add issue" workflow for a project table to add issues from one repo. This is a significant limitation and makes the auto-add feature of limited use for our work, as we have many repos that come together for a project. 

Our organization has many repos, and we would like to be able to auto-add issues from across all of our repos that meet certain search criteria on labels, or similar auto-add issues from across a subset of multiple repos.  

Similar to https://github.com/orgs/community/discussions/44826, but this request also asks for organization wide auto-add capability where has no repo-specific dependency, just the search criteria.

**Comment by kylekurz** [comment]
I'd like to +1 this request, particularly in light of a 20 auto-add workflow limit on the organization. We have dozens of repositories. Even managing one project would exhaust this limit. Getting projects up for all our PMs will require an ability to have a search filter per-PM instead of per-PM*per-repo.

**Comment by DFelten** [comment]
It's possible to duplicate the workflow. Unfortunately, you can currently only do this five times. 

Is there a plan to remove this limitation? We work on significantly more repositories in parallel and would therefore need this workflow more often. It would be even more practical if multiple repositories could be selected in one workflow.

**Reply by davetapley** [reply]
I'd hoped to work around this, but then: https://github.com/orgs/community/discussions/180056

**Comment by nfuzer** [comment]
+1 to this

**Comment by CarlOhlsson** [comment]
+1 to this, i have to monitor 25+ different repos for my company and this is a struggle

**Comment by rnhurt** [comment]
I really want to like GitHub Projects, but it's things like this that drive me away.  It seems like it's just not a tool that can be used for real projects.  

I'm not sure how GitHub uses Projects (or if they do) but I don't know how they avoided the problem of managing multiple repos in a single project.  Maybe they have a special version that has these limitations removed or something.  :shrug:

**Comment by slagelwa** [comment]
+1

**Comment by vc1492a** [comment]
+1 we're looking to create a team roadmap project we can use to help make decisions on workflows based on a backlog of tickets cross repositories and / or other projects.

**Reply by rnhurt** [reply]
We have a "roadmap" project but it's not easy syncing it with our "daily" project.  Good luck.

**Comment by lacraig2** [comment]
+1 this would be really helpful.

**Comment by smilefox-lab** [comment]
you can achieve a similar outcome by leveraging GitHub Actions in a custom workflow.

**Reply by lacraig2** [reply]
I saw that with https://github.com/marketplace/actions/add-to-github-projects

but it's a gross solution if I need to add that to every repo when that functionality that _almost_ exists in Github and just needs to adjusted to be org-wide.

**Reply by kpollich** [reply]
Adding a manually-maintained GitHub action across many repos has been painful. I'd love to see this feature embraced natively so I can delete these actions and their necessary secrets, etc.

**Comment by vc1492a** [comment]
> I saw that with https://github.com/marketplace/actions/add-to-github-projects
> 
> but it's a gross solution if I need to add that to every repo when that functionality that _almost_ exists in Github and just needs to adjusted to be org-wide.

Or team wide so that Teams within an org can make decisions for their workflows

**Comment by mosk-anw** [comment]
This is a **must-have requirement** to be able to map multiple repos to a single project so workflows can be triggered to create respective cards/tasks when issues/bugs are created in  the corresponding project, card view -  Kanban board, Roadmap , bug tracker view .

**Comment by bunnybones1** [comment]
The limits of 5 and 20 for Team and Enterprise respectively are frustratingly, arbitrarily low.

BUT, I added this to about 15 repos, with one Organization-level secret, and it works fine.
https://github.com/actions/add-to-project

Not ideal but it gets the job done.

**Comment by phifuh** [comment]
1 year later and this is still a crucial missing feature

**Comment by loewenstein-sap** [comment]
Yeah, things like this are actually "Let's keep using Jira" level missing features. Github Projects were looking very promising when they were in an early stage - haven't aged well I'd say.

**Comment by Piidgey** [comment]
+1 this is very much needed. We have 50+ repos and I need all new issues to go into the same projects board

**Comment by skeet70** [comment]
As an experiment today I was testing out Github Projects as a replacement for Zenhub for us, but the lack of being able to watch as many repos as desired likely kills that experiment. Our main board involves 33 repos, using an action added to each one to get around an arbitrary limit would be a huge pain.

**Reply by rnhurt** [reply]
> As an experiment today I was testing out Github Projects as a replacement for Zenhub for us, but the lack of being able to watch as many repos as desired likely kills that experiment. Our main board involves 33 repos, using an action added to each one to get around an arbitrary limit would be a huge pain.

Yes, we moved the GH Projects 18 months ago and so far it's been pretty terrible.  Sure it does some basic things but anything approaching "advanced" is not possible.  Handling multiple repos is a huge PITA; that seems like something GHP should excel at.  :shrug:

**Comment by thomas-bc** [comment]
This would really be a great improvement!

**Comment by MichaelKreil** [comment]
Since this feature is still missing, I wrote a little hack in bash using `gh`, `jq` and `xargs`.
1.	Make sure you have the correct project permissions (read/write access to projects):
```bash
gh auth refresh -s project
```

2.	Use `gh search` to find all issues without a project, extract their URLs, and then add them to the project of your choice:
```bash
gh search issues --owner="$ORGANIZATION" --no-project --json=url --limit=100 \
   | jq -r '.[] | .url' \
   | xargs -I {} gh issue edit {} --add-project "$PROJECT_TITLE"
```

**Comment by joelveloso** [comment]
+1

**Comment by Christian-onsai** [comment]
+1 for this request!

**Comment by GregTitrent** [comment]
+1 000 000

**Comment by benthamite** [comment]
Since this is still not natively supported, I created some scripts that automate the process. In case they are useful to anyone, I have made them publicly available [here](https://github.com/benthamite/issues-to-projects).

**Comment by davidgwatkins** [comment]
Year two of +1's on this.

**Comment by deganmee** [comment]
# +1

I'm curious why this product feedback hasn't gotten the template response comment that is posted on most Product Feedback discussions. Is it in the wrong format?

**Comment by jochso** [comment]
Still not implemented. I can only duplicate 5 "auto adds workflows" which is not enough for my organization,

**Comment by Skarlso** [comment]
It's insane that this still doesn't exist and I have to actually use a github action to do this.

**Comment by Skarlso** [comment]
This is actually now possible:

<img width="370" height="249" alt="Screenshot 2025-08-18 at 8 20 02" src="https://github.com/user-attachments/assets/df82180c-1be7-4dbe-a821-7caf4f3cd7f5" />
<img width="1567" height="446" alt="Screenshot 2025-08-18 at 8 20 08" src="https://github.com/user-attachments/assets/8d0f2880-a48b-4eb2-abe9-e27d1f5ff3d2" />

**Reply by DFelten** [reply]
But only for 5 repositories, right?

**Reply by Skarlso** [reply]
Ah good point. Sorry. For my case, 5 is enough. But I understand that in many cases it just isn't.

**Reply by thomas-bc** [reply]
Do you have a Team or special subscription? I can't have more than 1 custom workflow

**Reply by Skarlso** [reply]
I don't. This is really base view. 🤔

**Reply by davetapley** [reply]
Took me a moment to figure out: you have to click the '...' to see the 'Duplicate' button:

<img width="492" height="116" alt="Screen Shot 2025-10-09 at 1 01 53 PM" src="https://github.com/user-attachments/assets/f3a1aeaa-c1be-47a7-a5c0-b790488847b9" />

**Reply by drewclifton** [reply]
I don't have this three-dot "..." next to my workflows, so I can't duplicate them.
<img width="348" height="381" alt="Screenshot 2025-12-24 at 8 20 55 AM" src="https://github.com/user-attachments/assets/f7cca92c-1d52-4b74-aba9-0b615ec1c4e2" />

**Reply by cewood** [reply]
@drewclifton the three-dots are only for the “Auto-add to project” rules, and you probably need to enable the first/default rule for it to be enabled.

**Comment by dticson** [comment]
Me and claude wrote a script to handle this - you can add this as a cron job Github Action as well if you like:
```
#!/usr/bin/env bash

# requires project permissions - gh auth refresh -s project

ORG=
PROJ=
DEBUG=true  # Set to true to print issue URLs at each step

# 1) Get all issue URLs already in the project, limit is 1000 for max on the api/cli 
echo "Step 1: Getting all issues currently in project $PROJ..."
already=$(
  gh project item-list "$PROJ" --limit 1000 --owner "$ORG" --format json \
  | jq -r '.items[] | select(.content.type=="Issue") | .content.url'
)
already_count=$(echo "$already" | grep -c '^https://' || true)
[[ -z "$already_count" ]] && already_count=0
echo "Found $already_count issues already in the project"
if [[ "$DEBUG" == "true" ]]; then
  echo "  Issues in project:"
  echo "$already" | sed 's/^/    /'
fi
echo ""

# 2) Get all open issues across the organization
echo "Step 2: Getting all open issues across the $ORG organization..."
all_issues=$(
  gh search issues --owner "$ORG" is:open --json url --jq '.[].url' --limit 1000
)
all_issues_count=$(echo "$all_issues" | grep -c '^https://' || true)
[[ -z "$all_issues_count" ]] && all_issues_count=0
echo "Found $all_issues_count open issues across the organization"
if [[ "$DEBUG" == "true" ]]; then
  echo "  All open issues:"
  echo "$all_issues" | sed 's/^/    /'
fi
echo ""

# 3) Calculate the difference (issues to be added)
echo "Step 3: Calculating which issues need to be added..."
to_add=$(
  comm -13 \
    <(printf "%s\n" "$already" | sort) \
    <(printf "%s\n" "$all_issues" | sort)
)
# Count only non-empty lines
to_add_count=$(echo "$to_add" | grep -c '^https://' || true)
[[ -z "$to_add_count" ]] && to_add_count=0
echo "Found $to_add_count issues that need to be added to the project"
if [[ "$DEBUG" == "true" ]]; then
  echo "  Issues to be added:"
  if [[ "$to_add_count" -eq 0 ]]; then
    echo "    (none)"
  else
    echo "$to_add" | sed 's/^/    /'
  fi
fi
echo ""

# 4) Add the issues and print output
echo "Step 4: Adding issues to the project..."
if [[ -z "$to_add" || "$to_add_count" -eq 0 ]]; then
  echo "No issues to add. All open issues are already in the project!"
else
  echo "$to_add" | while IFS= read -r url; do
    [[ -z "$url" ]] && continue
    printf 'Adding %s ... ' "$url"
    if gh project item-add "$PROJ" --owner "$ORG" --url "$url" >/dev/null 2>&1; then
      echo "ADDED"
    else
      echo "ERROR"
    fi
  done
  echo ""
  echo "Done! Added issues to project."
fi
```

**Comment by Rnbsov** [comment]
oh my god, i wish so much it will be implemented 😢🥺

**Comment by Skyb0rg** [comment]
This would really be a great improvement! Please add this!!!

**Comment by gauravharsha** [comment]
+1 for this feature. Feels like we are in Stone Age without it.

**Comment by vizthex123** [comment]
100% agreed. Feels like it kinda defeats the point of the project system without this feature.

**Comment by ArnoClassenF19** [comment]
+1
